docs/DeveloperGuide.md

```diff
@@ -329,6 +333,49 @@ same state. For example, after `delete 1`, we might save the person just deleted
 functionality. However, if we do `add "person"`, the person will be added to the end of the list, which does not exactly
 `undo` the effect of the original `delete 1`.
 
+### Import CSV feature
+
+##### Aim of the feature
+
+CSV files are by far one of the most commonly used formats for organizing large
+amounts of data in an efficient manner. Since they are plain text files, they are
+easier to transfer onto a spreadsheet or another storage database regardless of
+the software being used. Since LumiNUS provides a way to import student data from CSV files,
+the user would not have to go through the repeated, error-prone process of
+manually adding each student's data to the database. `import-csv` automates this process and provides
+a one-command solution to this problem.
+
+#### Implementation
+
+The following classes were created in the process of implementing the `import-csv`
+command :
+
+| Logic                  | Commons |
+|------------------------|---------|
+| ImportCsvCommand       | CsvUtil |
+| ImportCsvCommandParser |         |
+
+The external library used for reading CSV files is OpenCSV. OpenCSV supports
+a host of functions that allow one to manipulate and work with CSV files. The
+command utilizes the `CSVReader` from the library to read the files.
+
+One important assumption made while implementing the `import-csv` command is that
+the files are expected to conform to the format defined by LumiNUS (the learning management
+system used by NUS). This however, can be modified and extended to cater to other systems
+as well.
+
+ImportCsvCommand takes in the path argument for the CSV file, which is parsed to retrieve
+the contents and load them into TAilor's database. Exception handling has been done 
+and checks (file not found, io, csvvalidation etc)
+
+#### Design Considerations
+
+**Aspect:**
```

Comment:
Is this part supposed to be empty?

Reply:
no no, it was all a framework at that point, have completed it now:))